src/core/qgspointlocator.cpp

```diff
@@ -134,6 +134,103 @@ class QgsPointLocator_VisitorNearestVertex : public IVisitor
 };
 
 
+
+/**
+ * \ingroup core
+ * Helper class used when traversing the index looking for centroid - builds a list of matches.
+ * \note not available in Python bindings
+ * \since QGIS 3.12
+*/
+class QgsPointLocator_VisitorNearestCentroid : public IVisitor
+{
+  public:
+    QgsPointLocator_VisitorNearestCentroid( QgsPointLocator *pl, QgsPointLocator::Match &m, const QgsPointXY &srcPoint, QgsPointLocator::MatchFilter *filter = nullptr )
+      : mLocator( pl )
+      , mBest( m )
+      , mSrcPoint( srcPoint )
+      , mFilter( filter )
+    {}
+
+    void visitNode( const INode &n ) override { Q_UNUSED( n ); }
+    void visitData( std::vector<const IData *> &v ) override { Q_UNUSED( v ); }
+
+    void visitData( const IData &d ) override
+    {
+      QgsFeatureId id = d.getIdentifier();
+      QgsGeometry *geom = mLocator->mGeoms.value( id );
+
+      QgsPointXY pt = geom->centroid().asPoint();
+
+      QgsPointLocator::Match m( QgsPointLocator::Centroid, mLocator->mLayer, id, std::sqrt( mSrcPoint.sqrDist( pt ) ), pt, -1 );
+      // in range queries the filter may reject some matches
+      if ( mFilter && !mFilter->acceptMatch( m ) )
+        return;
+
+      if ( !mBest.isValid() || m.distance() < mBest.distance() )
+        mBest = m;
+
+    }
+
+  private:
+    QgsPointLocator *mLocator = nullptr;
+    QgsPointLocator::Match &mBest;
+    QgsPointXY mSrcPoint;
+    QgsPointLocator::MatchFilter *mFilter = nullptr;
+};
+
+////////////////////////////////////////////////////////////////////////////
+
+/**
+ * \ingroup core
+ * Helper class used when traversing the index looking for middle segment - builds a list of matches.
+ * \note not available in Python bindings
+ * \since QGIS 3.12
+*/
+class QgsPointLocator_VisitorNearestMiddle: public IVisitor
```

Comment:
```suggestion
class QgsPointLocator_VisitorNearestMiddleSegment: public IVisitor
```

I don't find "middle" particularly explanatory -- could you use MiddleOfSegment instead throughout?

Reply:
You're right. done thanks.